### PR TITLE
Added query arrays tests for $size

### PR DESCRIPTION
**Pull request by vic-tsang**
This PR contains
- query arrays tests for $size

Ref:
- Issue: #30

### CODE REVIEW

File: documentdb_tests/compatibility/tests/core/operator/query/arrays/size/test_size_basic_matching.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+import pytest
+
+from documentdb_tests.compatibility.tests.core.operator.query.utils.query_test_case import (
+    QueryTestCase,
+)
+from documentdb_tests.framework.assertions import assertSuccess
+from documentdb_tests.framework.executor import execute_command
+from documentdb_tests.framework.parametrize import pytest_params
+
```
Comment (manojnks) [inline documentdb_tests/compatibility/tests/core/operator/query/arrays/size/test_size_basic_matching.py:16]: - Duplicate elements: [1, 1, 1] with $size: 3 should match. Not explicitly tested — important because some naïve implementations might count distinct values.                                                                                  
  - `[undefined]` element — in BSON, undefined is a distinct type from null. Worth testing.                                                                                                                                                      
  - Very large array at BSON size limit (~16MB). Impractical to test but worth mentioning. At minimum, test a 1000- or 10000-element array.

File: documentdb_tests/compatibility/tests/core/operator/query/arrays/size/test_size_errors.py [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+    TS_EPOCH,
+)
+
+DOCS = [
+    {"_id": 1, "a": []},
+    {"_id": 2, "a": [1]},
+    {"_id": 3, "a": [1, 2]},
+    {"_id": 4, "a": [1, 2, 3]},
+]
+
```
Comment (manojnks) [inline documentdb_tests/compatibility/tests/core/operator/query/arrays/size/test_size_errors.py:40]: - `$size` argument as a `javascript` BSON type — edge case but a valid BSON type that might produce a distinct error path.                                                                                                                     
  - Deeply-nested invalid arg: {a: {$size: [[1]]}} — nested array.                                                                                                                                                                               
  - `$size` with `$where` combinations — should error.

File: documentdb_tests/compatibility/tests/core/operator/query/arrays/size/test_size_field_lookup.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+import pytest
+
+from documentdb_tests.compatibility.tests.core.operator.query.utils.query_test_case import (
+    QueryTestCase,
+)
+from documentdb_tests.framework.assertions import assertSuccess
+from documentdb_tests.framework.executor import execute_command
+from documentdb_tests.framework.parametrize import pytest_params
+
```
Comment (manojnks) [inline documentdb_tests/compatibility/tests/core/operator/query/arrays/size/test_size_field_lookup.py:17]: - `$size` on array where the dotted path resolves to an array of arrays: {a: [[[1,2]]]} with a.0.0: {$size: 2} — boundary of how far dot-traversal goes.                                                                                                                  
- Dotted path where intermediate is null: {a: {b: null}} with a.b.c: {$size: 0} — does it error or just return no match? Not covered.

File: documentdb_tests/compatibility/tests/core/operator/query/arrays/size/test_size_query_contexts.py [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+        expected=[
+            {"_id": 2, "a": [1]},
+            {"_id": 3, "a": "x"},
+            {"_id": 4, "b": 1},
+        ],
+        msg="$not $size 0 matches non-empty arrays and non-array fields",
+    ),
+]
+
+COMBINATION_ARRAY_OPS_TESTS: list[QueryTestCase] = [
```
Comment (manojnks) [inline documentdb_tests/compatibility/tests/core/operator/query/arrays/size/test_size_query_contexts.py:16]: should combination test be present in this folder?
i believe it should in common folder.

File: documentdb_tests/compatibility/tests/core/operator/query/arrays/size/test_size_query_contexts.py [hunk 5]
```diff
@@ -0,0 +1,316 @@
...
+@pytest.mark.parametrize("test", pytest_params(SUCCESS_TESTS))
+def test_size_query_contexts(collection, test):
+    """Parametrized test for $size query contexts."""
+    if test.doc:
+        collection.insert_many(test.doc)
+    result = execute_command(collection, {"find": collection.name, "filter": test.filter})
+    assertSuccess(result, test.expected, ignore_doc_order=True)
+
+
+def test_size_in_aggregate_match(collection):
```
Comment (manojnks) [inline documentdb_tests/compatibility/tests/core/operator/query/arrays/size/test_size_query_contexts.py]: this should also be present in different folder since its a testing in a stage

File: documentdb_tests/compatibility/tests/core/operator/query/arrays/size/test_size_errors.py (hunk 2 above)
Comment (vic-tsang) [reply documentdb_tests/compatibility/tests/core/operator/query/arrays/size/test_size_errors.py:40]: add the first two. 
$size with $where combinations doesn't error. I can still that test as a combination test instead.

File: documentdb_tests/compatibility/tests/core/operator/query/arrays/size/test_size_basic_matching.py (hunk 1 above)
Comment (vic-tsang) [reply documentdb_tests/compatibility/tests/core/operator/query/arrays/size/test_size_basic_matching.py:16]: added duplicate elements and changed array size from 100 to 10000. For [undefined], there is no direct way to use that type in PyMongo.

File: documentdb_tests/compatibility/tests/core/operator/query/arrays/size/test_size_field_lookup.py (hunk 3 above)
Comment (vic-tsang) [reply documentdb_tests/compatibility/tests/core/operator/query/arrays/size/test_size_field_lookup.py:17]: added both. Dotted path where intermediate is null returns no match

File: documentdb_tests/compatibility/tests/core/operator/query/arrays/size/test_size_query_contexts.py (hunk 4 above)
Comment (vic-tsang) [reply documentdb_tests/compatibility/tests/core/operator/query/arrays/size/test_size_query_contexts.py:16]: move to test_query_combination_arrays_operators.py

**Comment by vic-tsang** [comment]
fixed reviewer comments and rebased

**Comment by manojnks** [comment]
aggregation is still  in the same file. is that okay?

**Review by manojnks — APPROVED** [review]

File: documentdb_tests/compatibility/tests/core/operator/query/arrays/size/test_size_query_contexts.py [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+import pytest
+
+from documentdb_tests.compatibility.tests.core.operator.query.utils.query_test_case import (
+    QueryTestCase,
+)
+from documentdb_tests.framework.assertions import assertSuccess
+from documentdb_tests.framework.executor import execute_command
+from documentdb_tests.framework.parametrize import pytest_params
+
+NEGATION_TESTS: list[QueryTestCase] = [
```
Comment (eerxuan) [inline documentdb_tests/compatibility/tests/core/operator/query/arrays/size/test_size_query_contexts.py]: These are combination tests or container test, do we need them?

File: documentdb_tests/compatibility/tests/core/operator/query/arrays/size/test_size_query_contexts.py (hunk 5 above)
Comment (vic-tsang) [reply documentdb_tests/compatibility/tests/core/operator/query/arrays/size/test_size_query_contexts.py]: removed as this test already exists In the match tests

File: documentdb_tests/compatibility/tests/core/operator/query/arrays/size/test_size_query_contexts.py (hunk 6 above)
Comment (vic-tsang) [reply documentdb_tests/compatibility/tests/core/operator/query/arrays/size/test_size_query_contexts.py]: removed file

**Review by eerxuan — APPROVED** [review]
